app/src/androidTest/java/org/mozilla/focus/activity/LocalSessionStorageTest.java

```diff
@@ -0,0 +1,165 @@
+        @Override
+        protected void beforeActivityLaunched() {
+            super.beforeActivityLaunched();
+
+            final Context appContext = InstrumentationRegistry.getInstrumentation()
+                    .getTargetContext()
+                    .getApplicationContext();
+
+            PreferenceManager.getDefaultSharedPreferences(appContext)
+                    .edit()
+                    .putBoolean(FIRSTRUN_PREF, true)
+                    .apply();
+
+            webServer = new MockWebServer();
+
+            try {
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("storage_start.html")));
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("storage_check.html")));
+                webServer.enqueue(new MockResponse()
+                        .setBody(TestHelper.readTestAsset("storage_check.html")));
+                webServer.start();
+            } catch (IOException e) {
+                throw new AssertionError("Could not start web server", e);
+            }
+        }
+
+        @Override
+        protected void afterActivityFinished() {
+            super.afterActivityFinished();
+
+            try {
+                 webServer.shutdown();
+            } catch (IOException e) {
+                throw new AssertionError("Could not stop web server", e);
+            }
+        }
+    };
+
+    /**
+     * Make sure that session storage values are kept and written but removed at the end of a session.
+     */
+    @Test
+    public void testLocaleAndSessionStorageIsWrittenAndRemoved() throws Exception {
+        // Go to first website that saves a value in the session/locale store.
+
+        goToUrlFromHomeScreen("/sessionStorage_start");
+
+        // Assert website is loaded and values are written.
+
+        final UiObject savedMsg = TestHelper.mDevice.findObject(new UiSelector()
+                .description("Values written to storage")
+                .enabled(true));
+        savedMsg.waitForExists(waitingTime);
+        assertTrue("Website loaded and values written to local / session storage", savedMsg.exists());
+
+        // Now load the next website and assert that the values are still in the storage
+
+        goToUrlFromBrowserScreen("/sessionStorage_check");
+
+        { // Value is still stored in session storage
+            final UiObject sessionStorageMsg = TestHelper.mDevice.findObject(new UiSelector()
+                    .description("Session storage has value")
+                    .enabled(true));
+            sessionStorageMsg.waitForExists(waitingTime);
+            assertTrue("Session storage has value", sessionStorageMsg.exists());
+        }
+        { // Local storage is empty
+            final UiObject localStorageMsg = TestHelper.mDevice.findObject(new UiSelector()
+                    .description("Local storage empty")
```

Comment:
\[PMD\] INFO: The String literal "Local storage empty" appears 4 times in this file; the first occurrence is on line 108